### DISCUSSION TITLE
Compiling mPython to an Executable, RP2040, ESP32

### DISCUSSION THREAD

**Discussion by nnitch** [micropython/micropython#11799] [General]
Hello,

I've had a look around and can't see anyone else asking the same question, so here goes.

Is it possible to build a MicroPython project into an executable (e.g. UF2) that can be flashed to my microcontroller (RP2040)?

It just makes things a lot easier once my project moves from a development stage, requiring frequent bug fixes, to a implemented stage, only requires an update when a new feature is released, I just need to re-flash the firmware.

Cheers

**Comment by peterhinch** [comment]
See [frozen bytecode](http://docs.micropython.org/en/latest/develop/optimizations.html). tl;dr You can create a build containing all your MP modules. To make the MP code run at boot requires `main.py` to kick it off, typically with
```python
import my_main_module
```
I'm not sure if current MP allows `main.py` to be frozen, but if not there is a simple solution.

**Reply by nnitch** [reply]
Cheers :)
